rotkehlchen/chain/ethereum/modules/convex/convex_cache.py

```diff
@@ -0,0 +1,135 @@
+    return {
+        string_to_evm_address(address) for address in globaldb_get_general_cache_values(
+            cursor=cursor,
+            key_parts=(CacheType.CONVEX_POOL_ADDRESS,),
+        )
+    }
+
+
+def save_convex_data_to_cache(
+        write_cursor: DBCursor,
+        database: 'DBHandler',  # pylint: disable=unused-argument
+        new_data: list[ConvexPoolData],
+) -> None:
+    """Stores data about convex pools and their names in the global db cache tables."""
+    for pool in new_data:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.CONVEX_POOL_ADDRESS,),
+            values=[pool.pool_address],
+        )
+        globaldb_set_unique_cache_value(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.CONVEX_POOL_NAME, pool.pool_address),
+            value=pool.pool_name,
+        )
+
+
+def read_convex_data_from_cache() -> dict[ChecksumEvmAddress, str]:
+    """Reads convex pools and names from global db cache tables."""
+    pools: dict[ChecksumEvmAddress, str] = {}
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        for pool_address in get_existing_pools(cursor):
+            if (pool_name := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.CONVEX_POOL_NAME, pool_address),
+            )) is None:
+                continue
+            pools[pool_address] = pool_name
+
+    return pools
+
+
+def query_convex_data_from_chain(ethereum: 'EthereumInquirer') -> dict[ChecksumEvmAddress, str]:
+    """
+    Query Booster contract and fetch all reward pools.
+    May raise:
+    - RemoteError if failed to query chain
+    """
+    booster_contract = ethereum.contracts.contract(BOOSTER)
+    pools_count = booster_contract.call(
+        node_inquirer=ethereum,
+        method_name='poolLength',
+    )
+    calls_to_booster = [(
+        booster_contract.address,
+        booster_contract.encode('poolInfo', [x]),
+    ) for x in range(pools_count)]
+    booster_result = ethereum.multicall(
+        calls=calls_to_booster,
+    )
+    convex_rewards_addrs: list[ChecksumEvmAddress] = []
+    convex_lp_tokens_addrs = []
+    lp_tokens_contract = EvmContract(  # only need it to encode and decode
+        address=ZERO_ADDRESS,
+        abi=ethereum.contracts.abi('CONVEX_LP_TOKEN'),
+        deployed_block=0,
+    )
+    for single_booster_result in booster_result:
+        lp_token_addr = hex_or_bytes_to_address(single_booster_result[0:32])
+        crv_rewards = hex_or_bytes_to_address(single_booster_result[3 * 32:4 * 32])
+        convex_rewards_addrs.append(crv_rewards)
+        convex_lp_tokens_addrs.append(lp_token_addr)
+
+    calls_to_lp_tokens = [(lp_token_addr, lp_tokens_contract.encode('symbol')) for lp_token_addr in convex_lp_tokens_addrs]  # noqa: E501
+    lp_tokens_result = ethereum.multicall(
+        calls=calls_to_lp_tokens,
+    )
```

Comment:
Move the contract creation nearer to where it is used

```suggestion
    for single_booster_result in booster_result:
        lp_token_addr = hex_or_bytes_to_address(single_booster_result[0:32])
        crv_rewards = hex_or_bytes_to_address(single_booster_result[3 * 32:4 * 32])
        convex_rewards_addrs.append(crv_rewards)
        convex_lp_tokens_addrs.append(lp_token_addr)

    lp_tokens_contract = EvmContract(  # only need it to encode and decode
        address=ZERO_ADDRESS,
        abi=ethereum.contracts.abi('CONVEX_LP_TOKEN'),
        deployed_block=0,
    )
    calls_to_lp_tokens = [(lp_token_addr, lp_tokens_contract.encode('symbol')) for lp_token_addr in convex_lp_tokens_addrs]  # noqa: E501
    lp_tokens_result = ethereum.multicall(calls=calls_to_lp_tokens)
```